src/ui/zcl_abapgit_gui_css_processor.clas.abap

```diff
@@ -0,0 +1,116 @@
+              RAISING   zcx_abapgit_exception.
+  PROTECTED SECTION.
+  PRIVATE SECTION.
+    TYPES:
+      BEGIN OF gty_css_var,
+        name  TYPE string,
+        value TYPE string,
+      END OF gty_css_var,
+      gty_css_var_tab TYPE SORTED TABLE OF gty_css_var WITH UNIQUE KEY name.
+    METHODS:
+      get_css_vars_in_string IMPORTING iv_string           TYPE string
+                             RETURNING VALUE(rt_variables) TYPE gty_css_var_tab.
+    DATA:
+      mi_asset_manager TYPE REF TO zif_abapgit_gui_asset_manager,
+      mt_files         TYPE STANDARD TABLE OF string.
+ENDCLASS.
+
+
+
+CLASS zcl_abapgit_gui_css_processor IMPLEMENTATION.
+  METHOD constructor.
+    mi_asset_manager = ii_asset_manager.
+  ENDMETHOD.
+
+  METHOD add_file.
+    APPEND iv_url TO mt_files.
+  ENDMETHOD.
+
+  METHOD process.
+    DATA: ls_asset            TYPE zif_abapgit_gui_asset_manager=>ty_web_asset,
+          lt_contents         TYPE STANDARD TABLE OF string,
+          lv_content          TYPE string,
+          lt_css_variables    TYPE gty_css_var_tab,
+          lt_css_vars_in_file TYPE gty_css_var_tab.
+    FIELD-SYMBOLS: <lv_url>                TYPE string,
+                   <ls_css_variable>       LIKE LINE OF lt_css_vars_in_file,
+                   <ls_other_css_variable> LIKE LINE OF lt_css_variables,
+                   <lv_content>            LIKE LINE OF lt_contents.
+
+    LOOP AT mt_files ASSIGNING <lv_url>.
+      ls_asset = mi_asset_manager->get_asset( <lv_url> ).
+      ASSERT ls_asset-type = 'text' AND ls_asset-subtype = 'css'.
+      lv_content = zcl_abapgit_convert=>xstring_to_string_utf8( ls_asset-content ).
+
+      lt_css_vars_in_file = get_css_vars_in_string( lv_content ).
+
+      LOOP AT lt_css_vars_in_file ASSIGNING <ls_css_variable>.
+        INSERT <ls_css_variable> INTO TABLE lt_css_variables.
+        IF sy-subrc <> 0.
+          MODIFY TABLE lt_css_variables FROM <ls_css_variable>.
+        ENDIF.
+      ENDLOOP.
+
+      APPEND lv_content TO lt_contents.
+    ENDLOOP.
+
+    " 2. Replace all variable usages in variables
+    " TODO: this only handles one level of referencing
+    LOOP AT lt_css_variables ASSIGNING <ls_css_variable>.
+      LOOP AT lt_css_variables ASSIGNING <ls_other_css_variable> WHERE name <> <ls_css_variable>-name.
+        REPLACE ALL OCCURRENCES OF |var(--{ <ls_other_css_variable>-name })|
+                IN <ls_css_variable>-value
+                WITH <ls_other_css_variable>-value.
+      ENDLOOP.
+    ENDLOOP.
+
+    " 3. Replace all other variable usages by inlining the values.
+    LOOP AT lt_contents ASSIGNING <lv_content>.
+      LOOP AT lt_css_variables ASSIGNING <ls_css_variable>.
+        REPLACE ALL OCCURRENCES OF |var(--{ <ls_css_variable>-name })|
+                IN <lv_content>
+                WITH <ls_css_variable>-value.
+      ENDLOOP.
+    ENDLOOP.
+
+    rv_result = concat_lines_of( table = lt_contents sep = cl_abap_char_utilities=>cr_lf ).
```

Comment:
maybe newline better ? mostly newline char is used over the AG files from what I remember